stable_baselines3/common/torch_layers.py

```diff
@@ -128,20 +131,48 @@ def create_mlp(
     :param squash_output: Whether to squash the output using a Tanh
         activation function
     :param with_bias: If set to False, the layers will not learn an additive bias
-    :return:
+    :param pre_linear_modules: List of nn.Module to add before the linear layers,
+        for instance, BatchNorm layers.
+        Compared to post_linear_modules, they are used before the output layer (output_dim > 0).
+        The number of input features is passed to the module's constructor.
+    :param post_linear_modules: List of nn.Module to add after the linear layers (and before the activation function),
+        for instance, Dropout or LayerNorm layers.
+        They are not used after the output layer (output_dim > 0).
+        The number of input features is passed to the module's constructor.
+    :return: The list of layers of the neural network
     """
 
+    pre_linear_modules = pre_linear_modules or []
+    post_linear_modules = post_linear_modules or []
+
+    modules = []
     if len(net_arch) > 0:
-        modules = [nn.Linear(input_dim, net_arch[0], bias=with_bias), activation_fn()]
-    else:
-        modules = []
+        for module in pre_linear_modules:
+            modules.append(module(input_dim))
```

Comment:
Is the input dim the same for all modules?

Reply:
I think I get it. It only allows modules that have the same input/output dimension.

Reply:
maybe it is clearer with this test: https://github.com/DLR-RM/stable-baselines3/blob/3b84f71bb2b47ca2dc657c7c5205901b6c061f30/tests/test_custom_policy.py#L102-L104

shall I add a comment to avoid confusion?

Reply:
Yes! I've also suggested a new documentation: https://github.com/DLR-RM/stable-baselines3/pull/1975/files#r1686213647